openmdao/components/submodel_comp.py

```diff
@@ -75,6 +82,27 @@ def __init__(self, problem, inputs=None, outputs=None, reports=False, **kwargs):
                 else:
                     raise Exception(f'Expected output of type str or tuple, got {type(out)}.')
 
+    def set_subprob_solver_print(self, level=2, depth=1e99, type_='all'):
```

Comment:
So I think it would be preferable for the user to interact with the subprob in this case rather than saving these settings.

Let's not introduce public functions if they're not necessary.  The user already has to instantiate this component with a problem instance and they therefore have it available at their disposal.  Let them call `set_solver_print` on that problem, and by removing the hard-coded `subprob.set_solver_print(-1)` below, anything the user does there should be respected.

Reply:
got rid of public function